src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -280,14 +280,34 @@ else if (module == RegexpSinglelineCheck.class
      */
     public static Object getFieldValue(Field field, Object instance)
             throws MacroExecutionException {
+        Object returnValue = null;
+
         try {
-            // required for package/private classes
-            field.trySetAccessible();
-            return field.get(instance);
+            if (field != null) {
+                // required for package/private classes
+                field.trySetAccessible();
+                returnValue = field.get(instance);
+            }
+            else {
+                final Set<String> properties =
+                        SiteUtil.getPropertiesForDocumentation(instance.getClass(), instance);
+                final String fullClassName = instance.getClass().getName();
+                final String className =
+                        fullClassName.substring(fullClassName.lastIndexOf('.') + 1);
+                for (String property : properties) {
+                    final String formattedString = className + DOT + property;
+
+                    if (PROPERTIES_ALLOWED_GET_TYPES_FROM_METHOD.contains(formattedString)) {
+                        returnValue = null;
```

Comment:
location of this code is weird.

lets try:
```
public static String getDefaultValue(String propertyName, Field field,
                                         Object classInstance, String moduleName)
            throws MacroExecutionException {
 final Object value;
        if (field == null) {
             if (allowed_from_method) {
                    value = null
              }
       } esle {
        value = getFieldValue(field, classInstance);
}
```

Reply:
if field is null we should not execute ` getFieldValue(field, classInstance)` at all.

Reply:
> if field is null we should not execute ` getFieldValue(field, classInstance)` at all.

Will this work?
```
public static String getDefaultValue(String propertyName, Field field,
                                     Object classInstance, String moduleName)
        throws MacroExecutionException {
    final Object value;
    if (field != null) {
        value = getFieldValue(field, classInstance);
    } else {
        value = null;
    }
    
    ...
    }
```

I am confused about `if (allowed_from_method)`. As mentioned, if no field, then value = null. hence, i assume we don't need 
```
else {
                final Set<String> properties =
                        SiteUtil.getPropertiesForDocumentation(instance.getClass(), instance);
                final String fullClassName = instance.getClass().getName();
                final String className =
                        fullClassName.substring(fullClassName.lastIndexOf('.') + 1);
                for (String property : properties) {
                    final String formattedString = className + DOT + property;

                    if (PROPERTIES_ALLOWED_GET_TYPES_FROM_METHOD.contains(formattedString)) {
                        returnValue = null;
                        break;
                    }
                }
            }
```